ddtrace/internal/flare/flare.py

```diff
@@ -36,14 +35,28 @@ class FlareSendRequest:
     source: str = "tracer_python"
 
 
+class TracerFlareSendError(Exception):
+    def __init__(self, message: str):
+        super().__init__(message)
+        self.message = message
+
+
 class Flare:
-    def __init__(self, timeout_sec: int = DEFAULT_TIMEOUT_SECONDS, flare_dir: str = TRACER_FLARE_DIRECTORY):
+    def __init__(
+        self,
+        trace_agent_url: str,
+        api_key: Optional[str] = None,
+        timeout_sec: int = DEFAULT_TIMEOUT_SECONDS,
+        flare_dir: str = TRACER_FLARE_DIRECTORY,
+    ):
         self.original_log_level: int = logging.NOTSET
         self.timeout: int = timeout_sec
         self.flare_dir: pathlib.Path = pathlib.Path(flare_dir)
         self.file_handler: Optional[RotatingFileHandler] = None
+        self.url: str = trace_agent_url
+        self._api_key: Optional[str] = api_key
```

Comment:
If this is a concern, I can have the API key get passed into the `send()` method instead